### DISCUSSION TITLE
Inquiry about outputting the values of concentration of all the points in 2D CH model

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#20362] [Q&A Modules: Phase field] [answered]
Hi everyone,

Do you know how I can output all the points' concentration in a 2D CH model in each time step?
I know there is one command, called LineValueSampler.
However, from its introduction, it can only output 1D model's concentration, right? Because it can only output one line's variable's value, right? Can it output 2D model's concentration? Or, is there other command that can output all the points' concentration in 2D mode?
Does everyone have any good suggestion?

Thank you very much.
Best regards,

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can use this VectorPostprocessor (outputs to CSV) for all element values:
https://mooseframework.inl.gov/source/vectorpostprocessors/ElementValueSampler.html
or if you have a 2D array of points you can fit as a long parameter:
https://mooseframework.inl.gov/source/vectorpostprocessors/PointValueSampler.html

Guillaume

**Reply by abc-hy** [reply]
Hi, Guillaume,
Thank you very much for your quick response.
Do you mean that I need to write the command as shown below?

[VectorPostprocessors]
  [./element_value_sampler]
    type = ElementValueSampler
    variable = 'c'
    sort_by = id
  [../]
[]

How should I set the 'sort by' option? The 'sort by' option is set as id? And then I can get all the points' concentration? Will the csv file give me only one line and contains all of the points' concentration or it will be distributed by x axis and y axis?

**Reply by abc-hy** [reply]
And for your second suggestions, it seems that pointvaluesampler can only output several specific point's value, not the whole 2D matrix...

**Reply by GiudGiud** [reply]
For the second suggestion, you could input a 2D matrix of points, it ll just not be very pretty, a very long parameter essentially. 

I think the first suggestion should do what you want, if the Elements actually form the 2D matrix of points that you want

**Reply by GiudGiud** [reply]
~~Better answer to this is to define a nodeset then use the NodalValueSampler
or a new subdomain then the ElementValueSampler~~

**Reply by GiudGiud** [reply]
nevermind no need for defining a subset since you want ALL values. Answer remains the same.

**Comment by abc-hy** [comment]
Can you show me an example? How to use it?

**Reply by GiudGiud** [reply]
see the input files listed in the docs for this vector postprocessor
https://mooseframework.inl.gov/source/vectorpostprocessors/ElementValueSampler.html

**Comment by abc-hy** [comment]
[image: image.png]
In the example, it shows this. I feel confused about using this command?
Can you give me some advice?

**Reply by GiudGiud** [reply]
image isnt showing for me. Could you please copy paste the problematic extract?
